### PR TITLE
subsys: bootloader: Fix multi-image dfu for Network core

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Set the `signed_by_mcuboot_and_b0_${image_name}.bin` file instead of the `signed_by_b0_${image_name}.bin` file when Mcuboot works as a bootloader. Without that the multi-image dfu together with Mcuboot is not possible because the image is not signed by Mcuboot.

### CODE REVIEW

File: subsys/bootloader/cmake/packaging.cmake [hunk 1]
```diff
@@ -24,8 +24,13 @@ if(SB_CONFIG_DFU_MULTI_IMAGE_PACKAGE_BUILD)
   if(SB_CONFIG_DFU_MULTI_IMAGE_PACKAGE_NET)
     get_property(image_name GLOBAL PROPERTY DOMAIN_APP_CPUNET)
     list(APPEND dfu_multi_image_ids 1)
-    list(APPEND dfu_multi_image_paths "${PROJECT_BINARY_DIR}/signed_by_b0_${image_name}.bin")
-    list(APPEND dfu_multi_image_targets ${image_name}_extra_byproducts ${image_name}_signed_kernel_hex_target)
+    if(SB_CONFIG_BOOTLOADER_MCUBOOT)
+      list(APPEND dfu_multi_image_paths "${PROJECT_BINARY_DIR}/signed_by_mcuboot_and_b0_${image_name}.bin")
```
[the remark this answers is not in the document]
Comment (sigvartmh) [reply subsys/bootloader/cmake/packaging.cmake:28]: It shouldn't matter if it's signed by both but there will be some extra data appended to the image which probably is not needed so if you are squeezed on space for the application one could consider removing this.

**Comment by carlescufi** [comment]
DNM until @nordicjm  has reviewed it

File: subsys/bootloader/cmake/packaging.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (nordicjm) [reply subsys/bootloader/cmake/packaging.cmake:28]: b0 refers to network core too, so you are using b0 (though in other places it is b0n, it is still using the b0 key as there is only one key)

**Review by nordicjm — APPROVED** [review]

**Comment by ArekBalysNordic** [comment]
@nordicjm Do you know what's going on with mcuboot tests here?

**Comment by nordicjm** [comment]
I think that has been caused by something @tejlmand has done

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request merged**